apps/qa/src/pages/pluto/components/outlier_report.py

```diff
@@ -1,46 +1,85 @@
 import streamlit as st
 import pandas as pd
+from typing import Callable
 from st_aggrid import AgGrid
 
 
 class OutlierReport:
-    def __init__(self, data, v1, v2, condo, mapped):
+    def __init__(self, data, v, v_prev, condo, mapped):
         self.df = data
-        self.v1 = v1
-        self.v2 = v2
+        self.v = v
+        self.v_prev = v_prev
         self.condo = condo
         self.mapped = mapped
 
     def __call__(self):
         st.header("Outlier Analysis")
 
-        if self.v1 not in self.versions:
+        if self.v not in self.versions:
             st.info("There is no outlier report available for selected version.")
             return
 
-        self.display_dataframe("building_area_increase")
+        self.building_area_increase()
 
-        self.display_dataframe("unitsres_resarea")
+        self.small_apartments()
 
-        self.display_dataframe("lotarea_numfloor")
+        self.floors()
 
-    def display_dataframe(self, field):
-        df = self.fetch_dataframe(field)
+    def display_dataframe(
+        self, field: str, df: pd.DataFrame, filter: Callable | None = None
+    ):
+        st.markdown(self.markdown_dict[field])
+        st.info(self.info_dict[field])
 
         if df.empty:
-            st.markdown(self.markdown_dict[field])
             st.write("There are no outliers for this check.")
-            st.info(self.info_dict[field])
         else:
-            st.markdown(self.markdown_dict[field])
             st.write(f"There are {df.shape[0]} outliers in total.")
-            st.info(self.info_dict[field])
+            if filter:
+                df = filter(df)
 
-            with st.expander("See table"):
+            with st.expander("Show table"):
                 AgGrid(df)
 
+    def building_area_increase(self):
```

Comment:
Made these their own functions because of need for custom logic for filtering datasets